src/bp/core/services/action/action-service.ts

```diff
@@ -152,35 +162,66 @@ export class ScopedActionService {
     return !!actions.find(x => x.name === actionName)
   }
 
-  private _prepareRequire(fullPath: string) {
-    const actionLocation = path.dirname(fullPath)
+  async getActionDetails(actionName: string) {
+    const action = await this.findAction(actionName)
+    const code = await this.getActionScript(action)
+
+    const botFolder = action.location === 'global' ? 'global' : 'bots/' + this.botId
+    const dirPath = path.resolve(path.join(process.PROJECT_LOCATION, `/data/${botFolder}/actions/${actionName}.js`))
+    const lookups = getBaseLookupPaths(dirPath)
+
+    return { code, dirPath, lookups, action }
+  }
+
+  // This method tries to load require() files from the FS and fallback on BPFS
```

Comment:
Don't you think that we should try on BPFS then fallback to FS ? I mean it just make more sens to fallback on FS